cmd/thanosconvert/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/weaveworks/common/logging"
+	"gopkg.in/yaml.v2"
+
+	"github.com/cortexproject/cortex/pkg/storage/bucket"
+	"github.com/cortexproject/cortex/pkg/util/log"
+	"github.com/cortexproject/cortex/tools/thanosconvert"
+)
+
+func main() {
+	var (
+		configFilename string
+		dryRun         bool
+		cfg            bucket.Config
+	)
+
+	logfmt, loglvl := logging.Format{}, logging.Level{}
+	logfmt.RegisterFlags(flag.CommandLine)
+	loglvl.RegisterFlags(flag.CommandLine)
+	cfg.RegisterFlags(flag.CommandLine)
+	flag.StringVar(&configFilename, "config", "", "Path to bucket config YAML")
+	flag.BoolVar(&dryRun, "dry-run", false, "Don't make changes; only report what needs to be done")
+	flag.Usage = func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "%s is a tool to convert block metadata from Thanos to Cortex.\nPlease see %s for instructions on how to run it.\n\n", os.Args[0], "https://cortexmetrics.io/docs/blocks-storage/migrate-storage-from-thanos-and-prometheus/")
+		fmt.Fprintf(flag.CommandLine.Output(), "Flags:\n")
+		flag.PrintDefaults()
+	}
+	flag.Parse()
+
+	logger, err := log.NewPrometheusLogger(loglvl, logfmt)
+	if err != nil {
+		fmt.Printf("failed to create logger: %v\n", err)
+		flag.Usage()
```

Comment:
Printing usage hides the error message. I'd suggest removing flag.Usage() calls after print errors.